diff --git a/caffe2/CMakeLists.txt b/caffe2/CMakeLists.txt
index 0a419e4..833ad7e 100644
--- a/caffe2/CMakeLists.txt
+++ b/caffe2/CMakeLists.txt
@@ -1329,6 +1329,12 @@ target_include_directories(torch_cpu PRIVATE
 if(USE_KINETO)
   target_include_directories(torch_cpu PRIVATE
     ${TORCH_ROOT}/third_party/kineto/libkineto/src)
+
+  list(APPEND CMAKE_MODULE_PATH "/usr/local/musa/cmake")
+  find_package(MUSAToolkit REQUIRED)
+  list(POP_BACK CMAKE_MODULE_PATH)
+  get_target_property(MUSA_RT_LIB MUSA::musart IMPORTED_LOCATION)
+  target_link_libraries(torch_cpu PUBLIC ${MUSA_RT_LIB})
 endif()
 
 target_include_directories(torch_cpu PRIVATE
