diff --git a/cmake/Dependencies.cmake b/cmake/Dependencies.cmake
index 5227204..b24ed4b 100644
--- a/cmake/Dependencies.cmake
+++ b/cmake/Dependencies.cmake
@@ -1648,6 +1648,13 @@ if(USE_KINETO)
   message(STATUS "  KINETO_BUILD_TESTS = ${KINETO_BUILD_TESTS}")
   message(STATUS "  KINETO_LIBRARY_TYPE = ${KINETO_LIBRARY_TYPE}")
 
+  list(APPEND CMAKE_MODULE_PATH "/usr/local/musa/cmake")
+  find_package(MUSAToolkit REQUIRED)
+  list(POP_BACK CMAKE_MODULE_PATH)
+  get_target_property(MUPTI_INCLUDE_DIR MUSA::mupti INTERFACE_INCLUDE_DIRECTORIES)
+  get_target_property(MUSA_mupti_LIBRARY MUSA::mupti IMPORTED_LOCATION)
+  get_filename_component(MUSA_SOURCE_DIR ${MUPTI_INCLUDE_DIR} DIRECTORY)
+
   if(NOT LIBKINETO_NOCUPTI)
     set(CUDA_SOURCE_DIR "${CUDA_TOOLKIT_ROOT_DIR}" CACHE STRING "")
     message(STATUS "  CUDA_SOURCE_DIR = ${CUDA_SOURCE_DIR}")
