diff --git a/torch/csrc/profiler/collection.cpp b/torch/csrc/profiler/collection.cpp
index bd3bd3c..80ad36c 100644
--- a/torch/csrc/profiler/collection.cpp
+++ b/torch/csrc/profiler/collection.cpp
@@ -384,8 +384,10 @@ std::unique_ptr<KinetoObserverContext> ThreadLocalSubqueue::begin_op(
   if (config_.state == ProfilerState::KINETO_GPU_FALLBACK) {
     try {
       out->fallback_ = torch_ops_.device_fallback_.emplace_back();
-      torch::profiler::impl::cudaStubs()->record(
-          nullptr, &out->fallback_->device_event_start_, nullptr);
+      // musaStubs does not exist temporarily hence comment the following lines
+      // which won't affect the functionality of torch musa profiler
+      // torch::profiler::impl::cudaStubs()->record(
+      //     nullptr, &out->fallback_->device_event_start_, nullptr);
     } catch (const std::exception& e) {
       LOG(WARNING) << "Failed to record CUDA event. " << e.what();
     }
@@ -1110,8 +1112,8 @@ class TransferEvents {
                 TORCH_WARN_ONCE(
                     "ROCTracer produced duplicate flow start: ", i.flow.id);
               }
-#else // USE_ROCM
-              TORCH_INTERNAL_ASSERT(inserted.second);
+// #else // USE_ROCM
+//               TORCH_INTERNAL_ASSERT(inserted.second);
 #endif // USE_ROCM
             }
             TORCH_INTERNAL_ASSERT(e->parent_.expired());
