diff --git a/aten/src/ATen/native/EmbeddingBag.h b/aten/src/ATen/native/EmbeddingBag.h
index eb29e11..dc063e9 100644
--- a/aten/src/ATen/native/EmbeddingBag.h
+++ b/aten/src/ATen/native/EmbeddingBag.h
@@ -30,7 +30,7 @@ void check_arguments(
     const std::optional<Tensor>& per_sample_weights,
     bool include_last_offset);
 
-void make_bag_size_out(
+TORCH_API void make_bag_size_out(
     Tensor& bag_size_out,
     const Tensor& offsets,
     const Tensor& indices,
@@ -38,7 +38,7 @@ void make_bag_size_out(
     const bool include_last_offset,
     const bool requires_grad);
 
-void make_max_indices_out(
+TORCH_API void make_max_indices_out(
     Tensor& max_indices_out,
     const Tensor& weight,
     const Tensor& indices,
@@ -47,7 +47,7 @@ void make_max_indices_out(
     const int64_t mode,
     bool include_last_offset);
 
-void make_offset2bag_out(
+TORCH_API void make_offset2bag_out(
     Tensor& offset2bag,
     Tensor& output,
     const Tensor& weight,
