diff --git a/aten/src/ATen/native/quantized/AffineQuantizer.h b/aten/src/ATen/native/quantized/AffineQuantizer.h
index 1ff342a643c..f4b99667df3 100644
--- a/aten/src/ATen/native/quantized/AffineQuantizer.h
+++ b/aten/src/ATen/native/quantized/AffineQuantizer.h
@@ -8,37 +8,37 @@
 namespace at {
 namespace native {
 
-Tensor& quantize_tensor_per_tensor_affine(
+TORCH_API Tensor& quantize_tensor_per_tensor_affine(
     const Tensor& rtensor,
     Tensor& qtensor,
     double scale,
     int64_t zero_point);
-Tensor& quantize_tensor_per_channel_affine(
+TORCH_API Tensor& quantize_tensor_per_channel_affine(
     const Tensor& rtensor,
     Tensor& qtensor,
     Tensor scales,
     Tensor zero_points,
     int64_t axis);
 
-Tensor& quantize_tensor_per_channel_float_qparams(
+TORCH_API Tensor& quantize_tensor_per_channel_float_qparams(
     const Tensor& rtensor,
     Tensor& qtensor,
     Tensor scales,
     Tensor zero_points,
     int64_t axis);
 
-Tensor& dequantize_tensor_per_tensor_affine(
+TORCH_API Tensor& dequantize_tensor_per_tensor_affine(
     const Tensor& qtensor,
     Tensor& rtensor,
     double scale,
     int64_t zero_point);
-Tensor& dequantize_tensor_per_channel_affine(
+TORCH_API Tensor& dequantize_tensor_per_channel_affine(
     const Tensor& qtensor,
     Tensor& rtensor,
     Tensor scales,
     Tensor zero_points,
     int64_t axis);
-Tensor& dequantize_tensor_per_channel_float_qparams(
+TORCH_API Tensor& dequantize_tensor_per_channel_float_qparams(
     const Tensor& qtensor,
     Tensor& rtensor,
     Tensor scales,
