diff --git a/aten/src/ATen/native/layer_norm.cpp b/aten/src/ATen/native/layer_norm.cpp
index 37a3f1a..093bced 100644
--- a/aten/src/ATen/native/layer_norm.cpp
+++ b/aten/src/ATen/native/layer_norm.cpp
@@ -18,6 +18,8 @@
 #include <ATen/ops/native_layer_norm.h>
 #include <ATen/ops/native_layer_norm_backward_native.h>
 #include <ATen/ops/native_layer_norm_native.h>
+#include <ATen/ops/rms_norm_forward.h>
+#include <ATen/ops/rms_norm_native.h>
 #include <ATen/ops/zeros_like_native.h>
 #endif
 
@@ -259,5 +261,16 @@ std::tuple<Tensor, Tensor, Tensor> math_native_layer_norm(
   rstd = rstd.view(stat_shape);
   return std::make_tuple(out, mean, rstd);
 }
+
+at::Tensor rms_norm(
+    const at::Tensor& input,
+    at::IntArrayRef normalized_shape,
+    const c10::optional<at::Tensor>& weight_opt,
+    double eps) {
+  auto outputs_and_invvar =
+      at::rms_norm_forward(input, normalized_shape, weight_opt, eps);
+  return std::get<0>(outputs_and_invvar);
+}
+
 } // namespace native
 } // namespace at
