diff --git a/aten/src/ATen/cuda/CUDASparseDescriptors.cpp b/aten/src/ATen/cuda/CUDASparseDescriptors.cpp
index 092314ac81f..d7b029a0c52 100644
--- a/aten/src/ATen/cuda/CUDASparseDescriptors.cpp
+++ b/aten/src/ATen/cuda/CUDASparseDescriptors.cpp
@@ -22,8 +22,8 @@ void check_supported_cuda_type(cudaDataType cuda_type) {
   if (cuda_type == CUDA_R_16F) {
     cudaDeviceProp* prop = at::cuda::getCurrentDeviceProperties();
     TORCH_CHECK(
-        prop->major >= 5 && ((10 * prop->major + prop->minor) >= 53),
-        "Sparse operations with CUDA tensors of Float16 type are not supported on GPUs with compute capability < 5.3 (current: ",
+        prop->major >= 3 && ((10 * prop->major + prop->minor) >= 31),
+        "Sparse operations with CUDA tensors of Float16 type are not supported on GPUs with compute capability < 3.1 (current: ",
         prop->major,
         ".",
         prop->minor,
@@ -33,8 +33,8 @@ void check_supported_cuda_type(cudaDataType cuda_type) {
   if (cuda_type == CUDA_R_16BF) {
     cudaDeviceProp* prop = at::cuda::getCurrentDeviceProperties();
     TORCH_CHECK(
-        prop->major >= 8,
-        "Sparse operations with CUDA tensors of BFloat16 type are not supported on GPUs with compute capability < 8.0 (current: ",
+        prop->major >= 3,
+        "Sparse operations with CUDA tensors of BFloat16 type are not supported on GPUs with compute capability < 3.1 (current: ",
         prop->major,
         ".",
         prop->minor,
