diff --git a/aten/src/ATen/TensorIterator.h b/aten/src/ATen/TensorIterator.h
index 0e49151..abf274d 100644
--- a/aten/src/ATen/TensorIterator.h
+++ b/aten/src/ATen/TensorIterator.h
@@ -20,6 +20,10 @@ class OptionalTensorRef;
 using NameVector = SmallVector<Dimname, kDimVectorStaticSize>;
 } // namespace at
 
+namespace at::musa {
+class MusaTensorIterator;
+} // namespace at::musa
+
 // TensorIterator is a helper class for element-wise operations, such as
 // arithmetic, comparisons, and trigonometric functions. It handles
 // broadcasting and type conversions of operands.
@@ -239,6 +243,7 @@ enum class FastSetupType : uint8_t {
   NONE,
   CONTIGUOUS,
   CHANNELS_LAST,
+  CHANNELS_LAST_3D,
   NON_OVERLAPPING_DENSE
 };
 
@@ -784,6 +789,7 @@ class TORCH_API TensorIteratorConfig final {
  public:
   friend struct TensorIteratorBase;
   friend struct TensorIterator;
+  friend class musa::MusaTensorIterator;
 
   TensorIteratorConfig() = default;
 
