diff --git a/aten/src/ATen/Dispatch.h b/aten/src/ATen/Dispatch.h
index 5c7b39c..ddf9649 100644
--- a/aten/src/ATen/Dispatch.h
+++ b/aten/src/ATen/Dispatch.h
@@ -464,12 +464,12 @@ inline at::ScalarType scalar_type(at::ScalarType s) {
           SCALARTYPE6,                                  \
           __VA_ARGS__))
 
-#define AT_DISPATCH_CASE_INTEGRAL_TYPES(...)          \
-  AT_DISPATCH_CASE(at::ScalarType::Byte, __VA_ARGS__) \
-  AT_DISPATCH_CASE(at::ScalarType::Char, __VA_ARGS__) \
-  AT_DISPATCH_CASE(at::ScalarType::Int, __VA_ARGS__)  \
-  AT_DISPATCH_CASE(at::ScalarType::Long, __VA_ARGS__) \
-  AT_DISPATCH_CASE(at::ScalarType::Short, __VA_ARGS__)
+#define AT_DISPATCH_CASE_INTEGRAL_TYPES(...)           \
+  AT_DISPATCH_CASE(at::ScalarType::Long, __VA_ARGS__)  \
+  AT_DISPATCH_CASE(at::ScalarType::Int, __VA_ARGS__)   \
+  AT_DISPATCH_CASE(at::ScalarType::Short, __VA_ARGS__) \
+  AT_DISPATCH_CASE(at::ScalarType::Char, __VA_ARGS__)  \
+  AT_DISPATCH_CASE(at::ScalarType::Byte, __VA_ARGS__)
 
 #define AT_DISPATCH_INTEGRAL_TYPES(TYPE, NAME, ...) \
   AT_DISPATCH_SWITCH(TYPE, NAME, AT_DISPATCH_CASE_INTEGRAL_TYPES(__VA_ARGS__))
