diff --git a/aten/src/ATen/Dispatch.h b/aten/src/ATen/Dispatch.h
index b98d648..b8cadca 100644
--- a/aten/src/ATen/Dispatch.h
+++ b/aten/src/ATen/Dispatch.h
@@ -454,12 +454,12 @@ inline void deprecated_AT_DISPATCH_ALL_TYPES_AND_HALF_AND_COMPLEX() {}
           SCALARTYPE6,                                  \
           __VA_ARGS__))
 
-#define AT_DISPATCH_CASE_INTEGRAL_TYPES(...)          \
-  AT_DISPATCH_CASE(at::ScalarType::Byte, __VA_ARGS__) \
-  AT_DISPATCH_CASE(at::ScalarType::Char, __VA_ARGS__) \
-  AT_DISPATCH_CASE(at::ScalarType::Int, __VA_ARGS__)  \
-  AT_DISPATCH_CASE(at::ScalarType::Long, __VA_ARGS__) \
-  AT_DISPATCH_CASE(at::ScalarType::Short, __VA_ARGS__)
+#define AT_DISPATCH_CASE_INTEGRAL_TYPES(...)           \
+  AT_DISPATCH_CASE(at::ScalarType::Long, __VA_ARGS__)  \
+  AT_DISPATCH_CASE(at::ScalarType::Int, __VA_ARGS__)   \
+  AT_DISPATCH_CASE(at::ScalarType::Short, __VA_ARGS__) \
+  AT_DISPATCH_CASE(at::ScalarType::Char, __VA_ARGS__)  \
+  AT_DISPATCH_CASE(at::ScalarType::Byte, __VA_ARGS__)
 
 #define AT_DISPATCH_INTEGRAL_TYPES(TYPE, NAME, ...) \
   AT_DISPATCH_SWITCH(TYPE, NAME, AT_DISPATCH_CASE_INTEGRAL_TYPES(__VA_ARGS__))
