diff --git a/aten/src/ATen/native/LossNLL.cpp b/aten/src/ATen/native/LossNLL.cpp
index 0b07e79..002f584 100644
--- a/aten/src/ATen/native/LossNLL.cpp
+++ b/aten/src/ATen/native/LossNLL.cpp
@@ -26,6 +26,8 @@
 #include <ATen/ops/nll_loss_native.h>
 #include <ATen/ops/nll_loss_nd.h>
 #include <ATen/ops/nll_loss_nd_native.h>
+#include <ATen/ops/cross_entropy_loss_2d_choice.h>
+#include <ATen/ops/_fused_cross_entropy_loss_2d_forward.h>
 #endif
 
 #include <c10/core/TensorOptions.h>
@@ -646,6 +648,9 @@ Tensor cross_entropy_loss_symint(
     c10::MaybeOwned<Tensor> weight_maybe_owned = at::borrow_from_optional_tensor(weight);
     const Tensor& weight_ = *weight_maybe_owned;
     ret = cross_entropy_loss_label_smoothing(self, target, weight_, reduction, std::move(ignore_index), label_smoothing);
+  } else if ((self.dim() == 2) &&
+             (at::cross_entropy_loss_2d_choice(self, target, weight, reduction, ignore_index.expect_int(), label_smoothing) != 0)) {
+    ret = at::_fused_cross_entropy_loss_2d_forward(self, target, weight, reduction, ignore_index.expect_int(), label_smoothing);
   } else {
     auto class_dim = self.dim() == 1 ? 0 : 1;
     ret = at::nll_loss_nd_symint(
