diff --git a/aten/src/ATen/native/TensorAdvancedIndexing.cpp b/aten/src/ATen/native/TensorAdvancedIndexing.cpp
index d8d19af..869bc71 100644
--- a/aten/src/ATen/native/TensorAdvancedIndexing.cpp
+++ b/aten/src/ATen/native/TensorAdvancedIndexing.cpp
@@ -692,7 +692,8 @@ AdvancedIndex::AdvancedIndex(const Tensor& src, TensorList indices_list) {
   if (indices.size() >= 2 &&
       (this->src.device().type() == kCUDA ||
        this->src.device().type() == kMPS ||
-       this->src.device().type() == kXPU)) {
+       this->src.device().type() == kXPU ||
+       this->src.device().type() == kPrivateUse1)) {
     if (!all_strides_match(indices)) {
       for (auto& indice : indices) {
         indice = indice.contiguous();
