diff --git a/c10/core/DispatchKey.cpp b/c10/core/DispatchKey.cpp
index 526e7f079ee..01985255fb9 100644
--- a/c10/core/DispatchKey.cpp
+++ b/c10/core/DispatchKey.cpp
@@ -1,6 +1,7 @@
 #include <c10/core/DispatchKey.h>
 #include <c10/core/DispatchKeySet.h>
 
+#include <regex>
 #include <unordered_map>
 
 namespace c10 {
@@ -380,7 +381,10 @@ c10::DispatchKey parseDispatchKey(const std::string& k) {
       {"FuncTorchBatchedDecomposition",
        c10::DispatchKey::FuncTorchBatchedDecomposition},
   };
-  auto it = key_map.find(k);
+  std::string privateuse1_backend_name = c10::get_privateuse1_backend();
+  std::transform(privateuse1_backend_name.begin(), privateuse1_backend_name.end(), privateuse1_backend_name.begin(), ::toupper);
+  std::string processed_k = std::regex_replace(k, std::regex(privateuse1_backend_name), "PrivateUse1");
+  auto it = key_map.find(processed_k);
   TORCH_CHECK(it != key_map.end(), "could not parse dispatch key: ", k);
   return it->second;
 }
