diff --git a/c10/core/DispatchKey.cpp b/c10/core/DispatchKey.cpp
index d47de57..6b18c1c 100644
--- a/c10/core/DispatchKey.cpp
+++ b/c10/core/DispatchKey.cpp
@@ -372,7 +372,14 @@ c10::DispatchKey parseDispatchKey(const std::string& k) {
        c10::DispatchKey::FuncTorchBatchedDecomposition},
   };
   auto it = key_map.find(k);
-  TORCH_CHECK(it != key_map.end(), "could not parse dispatch key: ", k);
+  if (it == key_map.end()) {
+    std::string privateuse1_backend_name = c10::get_privateuse1_backend();
+    std::transform(privateuse1_backend_name.begin(), privateuse1_backend_name.end(), privateuse1_backend_name.begin(), ::toupper);
+    if (k == privateuse1_backend_name) {
+      return c10::DispatchKey::PrivateUse1;
+    }
+    TORCH_CHECK(false, "could not parse dispatch key: ", k);
+  }
   return it->second;
 }
 
