diff --git a/c10/util/C++17.h b/c10/util/C++17.h
index 359774b..b9aca70 100644
--- a/c10/util/C++17.h
+++ b/c10/util/C++17.h
@@ -45,7 +45,7 @@ constexpr bool is_pod_v = is_pod<T>::value;
 
 namespace guts {
 
-#if defined(__cpp_lib_apply) && !defined(__CUDA_ARCH__) && !defined(__HIP__)
+#if 0
 
 template <class F, class Tuple>
 C10_HOST_DEVICE inline constexpr decltype(auto) apply(F&& f, Tuple&& t) {
