diff --git a/c10/macros/Macros.h b/c10/macros/Macros.h
index 966a7a2..cba8da6 100644
--- a/c10/macros/Macros.h
+++ b/c10/macros/Macros.h
@@ -326,8 +326,7 @@ constexpr uint32_t CUDA_THREADS_PER_BLOCK_FALLBACK = 256;
 // CUDA_KERNEL_ASSERT checks the assertion
 // even when NDEBUG is defined. This is useful for important assertions in CUDA
 // code that would otherwise be suppressed when building Release.
-#if defined(__ANDROID__) || defined(__APPLE__) || \
-    (defined(USE_ROCM) && ROCM_VERSION < 40100)
+#if 1
 // Those platforms do not support assert()
 #define CUDA_KERNEL_ASSERT(cond)
 #define SYCL_KERNEL_ASSERT(cond)
