diff --git a/caffe2/CMakeLists.txt b/caffe2/CMakeLists.txt
index 9be7f37..fb02532 100644
--- a/caffe2/CMakeLists.txt
+++ b/caffe2/CMakeLists.txt
@@ -1192,6 +1192,15 @@ if(USE_KINETO)
     ${TORCH_ROOT}/third_party/kineto/libkineto/src)
 endif()
 
+if(DEFINED ENV{MUSA_HOME})
+  set(MUSA_HOME $ENV{MUSA_HOME})
+else()
+  set(MUSA_HOME "/usr/local/musa")
+endif()
+target_link_libraries(torch_cpu PUBLIC ${MUSA_HOME}/lib/libmupti.so)
+target_link_libraries(torch_cpu PUBLIC ${MUSA_HOME}/lib/libmusa.so)
+target_link_libraries(torch_cpu PUBLIC ${MUSA_HOME}/lib/libmusart.so)
+
 target_include_directories(torch_cpu PRIVATE
   ${TORCH_ROOT}/third_party/cpp-httplib)
 
