diff --git a/cmake/Dependencies.cmake b/cmake/Dependencies.cmake
index 8c0e3c2..cc9109a 100644
--- a/cmake/Dependencies.cmake
+++ b/cmake/Dependencies.cmake
@@ -1873,6 +1873,17 @@ if(USE_KINETO)
   message(STATUS "  KINETO_SOURCE_DIR = ${KINETO_SOURCE_DIR}")
   message(STATUS "  KINETO_BUILD_TESTS = ${KINETO_BUILD_TESTS}")
   message(STATUS "  KINETO_LIBRARY_TYPE = ${KINETO_LIBRARY_TYPE}")
+
+  if(DEFINED ENV{MUSA_HOME})
+    set(MUSA_HOME $ENV{MUSA_HOME})
+  else()
+    set(MUSA_HOME "/usr/local/musa")
+  endif()
+
+  set(MUSA_SOURCE_DIR ${MUSA_HOME} CACHE STRING "" FORCE)
+  set(MUPTI_INCLUDE_DIR ${MUSA_HOME}/include CACHE STRING "" FORCE)
+  set(MUSA_mupti_LIBRARY ${MUSA_HOME}/lib CACHE STRING "" FORCE)
+  set(LIBKINETO_NOMUPTI OFF CACHE STRING "" FORCE)

   if(NOT LIBKINETO_NOCUPTI)
     set(CUDA_SOURCE_DIR "${CUDA_TOOLKIT_ROOT_DIR}" CACHE STRING "")

