diff --git a/c10/core/Device.h b/c10/core/Device.h
index d53ab38..0d46a11 100644
--- a/c10/core/Device.h
+++ b/c10/core/Device.h
@@ -146,6 +146,11 @@ struct C10_API Device final {
     return type_ == DeviceType::CPU;
   }
 
+  /// Return true if the device is of PrivateUse1 type.
+  bool is_privateuseone() const noexcept {
+    return type_ == DeviceType::PrivateUse1;
+  }
+
   /// Return true if the device supports arbirtary strides.
   bool supports_as_strided() const noexcept {
     return type_ != DeviceType::IPU && type_ != DeviceType::XLA &&
