diff --git a/torch/distributed/distributed_c10d.py b/torch/distributed/distributed_c10d.py
index be0006d9ce..f1b111d965 100644
--- a/torch/distributed/distributed_c10d.py
+++ b/torch/distributed/distributed_c10d.py
@@ -241,6 +241,7 @@ class BackendConfig:
             self.device_backend_map = {
                 "cpu": backend_val,
                 "cuda": backend_val,
+                "musa": backend_val,
             }
         else:
             # custom backend string in format of "{device_type1}:{backend1},{device_type2}:{backend2}"
