diff --git a/aten/src/ATen/native/Dropout.cpp b/aten/src/ATen/native/Dropout.cpp
index 36e1b92ad1..0c68b1a3fe 100644
--- a/aten/src/ATen/native/Dropout.cpp
+++ b/aten/src/ATen/native/Dropout.cpp
@@ -25,8 +25,14 @@ Tensor make_feature_noise(const Tensor& input) {
   return input.new_empty(sizes);
 }
 
+// TODO(caizhi): A good implementation should be like "is_cuda()" in "Class Tensor" internal.
+// It is best not to modify this file directly, but through patching.
+bool is_mtgpu(const Tensor& input) {
+  return input.device().type() == DeviceType::MTGPU;
+}
+
 bool is_fused_kernel_acceptable(const Tensor& input, double p) {
-  return (input.is_cuda() || input.is_xpu() || input.is_lazy()) && p > 0 && p < 1 && input.numel() > 0;
+  return (input.is_cuda() || input.is_xpu() || input.is_lazy() || is_mtgpu(input)) && p > 0 && p < 1 && input.numel() > 0;
 }
 
 // NB: sure, we could have used different overloads here, but I would feel insecure
