diff --git a/torch/csrc/DynamicTypes.cpp b/torch/csrc/DynamicTypes.cpp
index 67357e4863..02f429c51d 100644
--- a/torch/csrc/DynamicTypes.cpp
+++ b/torch/csrc/DynamicTypes.cpp
@@ -154,6 +154,8 @@ at::Storage createStorageGetType(PyObject* obj, at::ScalarType& scalar_type, boo
     backend = at::Backend::CPU;
   } else if (device_type == at::kCUDA) {
     backend = at::Backend::CUDA;
+  } else if (device_type == at::kMTGPU) {
+    backend = at::Backend::MUSA;
   } else {
     TORCH_CHECK(false, "Invalid device for storage: ", device_type);
   }
