diff --git a/torch/csrc/DynamicTypes.cpp b/torch/csrc/DynamicTypes.cpp
index e818016202..357918fe07 100644
--- a/torch/csrc/DynamicTypes.cpp
+++ b/torch/csrc/DynamicTypes.cpp
@@ -45,6 +45,8 @@ at::DeprecatedTypeProperties* get_type_properties(
     backend = at::Backend::MPS;
   } else if (device_type == at::DeviceType::Meta) {
     backend = at::Backend::Undefined;
+  } else if (device_type == at::DeviceType::PrivateUse1) {
+    backend = at::Backend::PrivateUse1;
   } else {
     TORCH_CHECK(false, "Invalid device for storage: ", device_type);
   }
