diff --git a/torch/library.h b/torch/library.h
index 38887740ec..ebc61b1136 100644
--- a/torch/library.h
+++ b/torch/library.h
@@ -371,7 +371,7 @@ inline CppFunction dispatch(c10::DeviceType type, Func&& raw_f) {
         return c10::DispatchKey::ORT;
       case c10::DeviceType::HPU:
         return c10::DispatchKey::HPU;
-      case c10::DeviceType::PrivateUse1:
+      case c10::DeviceType::MTGPU:
         return c10::DispatchKey::PrivateUse1;
       default:
         TORCH_CHECK(
