diff --git a/setup.py b/setup.py
index 86cad76..008f241 100644
--- a/setup.py
+++ b/setup.py
@@ -1240,6 +1240,7 @@ def main():
         "include/torch/csrc/onnx/*.h",
         "include/torch/csrc/profiler/*.h",
         "include/torch/csrc/profiler/orchestration/*.h",
+        "include/torch/csrc/profiler/python/combined_traceback.h",
         "include/torch/csrc/profiler/stubs/*.h",
         "include/torch/csrc/profiler/unwind/*.h",
         "include/torch/csrc/utils/*.h",
