diff --git a/torch/csrc/Storage.cpp b/torch/csrc/Storage.cpp
index e998198..22f0489 100644
--- a/torch/csrc/Storage.cpp
+++ b/torch/csrc/Storage.cpp
@@ -112,6 +112,8 @@ static PyObject* THPStorage_pynew(
       allocator = c10::GetAllocator(device.type());
     } else if (device.type() == at::DeviceType::Meta) {
       allocator = c10::GetAllocator(device.type());
+    } else if (device.type() == at::DeviceType::PrivateUse1) {
+      allocator = c10::GetAllocator(device.type());
     } else {
       TORCH_CHECK(
           false,
