diff --git a/torch/csrc/generic/Storage.cpp b/torch/csrc/generic/Storage.cpp
index 83dfefa3d0..4cc7b6dbd7 100644
--- a/torch/csrc/generic/Storage.cpp
+++ b/torch/csrc/generic/Storage.cpp
@@ -67,6 +67,8 @@ static PyObject * THPStorage_(pynew)(PyTypeObject *type, PyObject *args, PyObjec
       at::globalContext().lazyInitCUDA();
       allocator = c10::cuda::CUDACachingAllocator::get();
 #endif
+    } else if (device.type() == at::kMTGPU) {
+      allocator = c10::GetAllocator(device.type());
     } else {
       TORCH_CHECK(false,
         THPStorageStr, "(): Storage device not recognized: ", device.type());
