diff --git a/aten/src/ATen/core/Tensor.cpp b/aten/src/ATen/core/Tensor.cpp
index fa175165d2..29c16dacd7 100644
--- a/aten/src/ATen/core/Tensor.cpp
+++ b/aten/src/ATen/core/Tensor.cpp
@@ -83,7 +83,16 @@ std::string TensorBase::toString() const {
   if (scalar_type() == ScalarType::Undefined) {
     base_str = "UndefinedType";
   } else {
-    base_str = std::string(at::toString(options().computeDispatchKey())) + at::toString(scalar_type()) + "Type";
+    auto dispatchkey = options().computeDispatchKey();
+    std::string dispatchkey_str;
+    if (dispatchkey == c10::DispatchKey::PrivateUse1) {
+      dispatchkey_str = c10::get_privateuse1_backend();
+    } else if (dispatchkey == c10::DispatchKey::QuantizedPrivateUse1) {
+      dispatchkey_str = "Quantized" + c10::get_privateuse1_backend();
+    } else {
+      dispatchkey_str = at::toString(dispatchkey);
+    }
+    base_str = dispatchkey_str + at::toString(scalar_type()) + "Type";
   }
   return base_str;
 }
