diff --git a/torch/csrc/utils/tensor_types.cpp b/torch/csrc/utils/tensor_types.cpp
index c52f10d172..e84e95f202 100644
--- a/torch/csrc/utils/tensor_types.cpp
+++ b/torch/csrc/utils/tensor_types.cpp
@@ -28,7 +28,7 @@ static const char* backend_to_string(const at::Backend& backend) {
     case at::Backend::QuantizedCPU: return "torch.quantized";
     case at::Backend::HPU: return "torch.hpu";
     case at::Backend::MPS: return "torch.mps";
-    case at::Backend::PrivateUse1: return "torch.privateuseone";
+    case at::Backend::MUSA: return "torch.musa";
     default: AT_ERROR("Unimplemented backend ", backend);
   }
 }
