diff --git a/c10/core/TensorOptions.h b/c10/core/TensorOptions.h
index e906720ba6..256e924eb2 100644
--- a/c10/core/TensorOptions.h
+++ b/c10/core/TensorOptions.h
@@ -683,9 +683,8 @@ inline DispatchKey computeDispatchKey(
           return DispatchKey::Meta;
         case DeviceType::HPU:
           return DispatchKey::HPU;
-        case DeviceType::PrivateUse1: {
+        case DeviceType::MTGPU:
           return DispatchKey::PrivateUse1;
-        }
         default:
           TORCH_CHECK_NOT_IMPLEMENTED(
               false,
@@ -794,6 +793,8 @@ inline DeviceType dispatchKeyToDeviceType(DispatchKey dispatch_key) {
       return DeviceType::Vulkan;
     case DispatchKey::Meta:
       return DeviceType::Meta;
+    case DispatchKey::PrivateUse1:
+      return DeviceType::MTGPU;
 
     // stuff that people are actively developing
     case DispatchKey::IPU:
@@ -812,8 +813,6 @@ inline DeviceType dispatchKeyToDeviceType(DispatchKey dispatch_key) {
       return DeviceType::HPU;
     case DispatchKey::ORT:
       return DeviceType::ORT;
-    case DispatchKey::PrivateUse1:
-      return DeviceType::PrivateUse1;
     default:
       TORCH_CHECK(
           false,
