diff --git a/torch/distributed/elastic/rendezvous/api.py b/torch/distributed/elastic/rendezvous/api.py
index cbfc553..38fdb8a 100644
--- a/torch/distributed/elastic/rendezvous/api.py
+++ b/torch/distributed/elastic/rendezvous/api.py
@@ -254,6 +254,7 @@ class RendezvousParameters:
         min_nodes: int,
         max_nodes: int,
         local_addr: Optional[str] = None,
+        use_libuv: bool = True,
         **kwargs,
     ):
         if not backend:
@@ -276,6 +277,7 @@ class RendezvousParameters:
         self.max_nodes = max_nodes
         self.config = kwargs
         self.local_addr = local_addr
+        self.use_libuv = use_libuv
 
     def get(self, key: str, default: Any = None) -> Any:
         """Return the value for ``key`` if ``key`` exists, else ``default``."""
@@ -365,7 +367,7 @@ class RendezvousHandlerRegistry:
         handler = creator(params)
 
         # Do some sanity check.
-        if handler.get_backend() != params.backend:
+        if params.backend != "c10d_v2" and handler.get_backend() != params.backend:
             raise RuntimeError(
                 f"The rendezvous backend '{handler.get_backend()}' does not match the requested "
                 f"backend '{params.backend}'."
