diff --git a/torch/profiler/profiler.py b/torch/profiler/profiler.py
index e0052c104f2..73cc2e6 100644
--- a/torch/profiler/profiler.py
+++ b/torch/profiler/profiler.py
@@ -226,6 +226,13 @@ class _KinetoProfile:
             )
         self.profiler._prepare_trace()

+        try:
+            torch.musa.update_musa_graph_with_profile()
+        except AttributeError:
+            # ignore the error if musa is not installed or
+            # update_musa_graph_with_profile is not ready in old version
+            pass
+
     def start_trace(self) -> None:
         if self.execution_trace_observer:
             self.execution_trace_observer.start()
