diff --git a/torch/csrc/utils/device_lazy_init.cpp b/torch/csrc/utils/device_lazy_init.cpp
index 74adb6b..702a4d2 100644
--- a/torch/csrc/utils/device_lazy_init.cpp
+++ b/torch/csrc/utils/device_lazy_init.cpp
@@ -32,7 +32,10 @@ void device_lazy_init(at::DeviceType device_type) {
     return;
   }
 
-  std::string module_name = "torch." + at::DeviceTypeName(device_type, true);
+  const auto device_type_name = at::DeviceTypeName(device_type, true);
+  std::string module_name = device_type == c10::kPrivateUse1
+      ? "torch_" + device_type_name
+      : "torch." + device_type_name;
   auto module = THPObjectPtr(PyImport_ImportModule(module_name.c_str()));
   if (!module) {
     throw python_error();
