diff --git a/torch/multiprocessing/reductions.py b/torch/multiprocessing/reductions.py
index f5eb0a6abd..4d46a6201a 100644
--- a/torch/multiprocessing/reductions.py
+++ b/torch/multiprocessing/reductions.py
@@ -308,7 +308,7 @@ def reduce_tensor(tensor):
 
     storage = tensor._typed_storage()
 
-    if storage._untyped_storage.device.type == "cuda":
+    if storage._untyped_storage.device.type in ["cuda", "musa"]:
         (
             device,
             handle,
@@ -553,6 +553,10 @@ def reduce_storage(storage):
         raise RuntimeError(
             "Cannot pickle CUDA storage; try pickling a CUDA tensor instead"
         )
+    elif storage.is_musa:
+        raise RuntimeError(
+            "Cannot pickle MUSA storage; try pickling a MUSA tensor instead"
+        )
     elif get_sharing_strategy() == "file_system":
         metadata = storage._share_filename_cpu_()
         cache_key = metadata[1]
