diff --git a/torch/csrc/inductor/aoti_runtime/model_container.h b/torch/csrc/inductor/aoti_runtime/model_container.h
index dc39bad..9923221 100644
--- a/torch/csrc/inductor/aoti_runtime/model_container.h
+++ b/torch/csrc/inductor/aoti_runtime/model_container.h
@@ -375,12 +375,12 @@ class AOTInductorModelContainer {
       queue_ptr
           ->memcpy(internal_constants_ptr, user_constant_ptr, constant_size)
           .wait();
-#elif USE_CUDA
-      AOTI_RUNTIME_DEVICE_CHECK(cudaMemcpy(
+#elif 1
+      AOTI_RUNTIME_DEVICE_CHECK(musaMemcpy(
           internal_constants_ptr,
           user_constant_ptr,
           constant_size,
-          cudaMemcpyDefault));
+          musaMemcpyDefault));
 #else
       memcpy(internal_constants_ptr, user_constant_ptr, constant_size);
 #endif
@@ -546,7 +546,7 @@ class AOTInductorModelContainer {
       return constant_blob_.get();
     } else {
       if (!constant_blob_secondary_) {
-#if defined(USE_CUDA) || defined(USE_XPU)
+#if 1 || defined(USE_XPU)
         constant_blob_secondary_ = RAII_gpuMalloc(blob_size_);
 #else
         constant_blob_secondary_ = RAII_cpuMalloc(blob_size_);
