diff --git a/torch/serialization.py b/torch/serialization.py
index d936d31..248ce2c 100644
--- a/torch/serialization.py
+++ b/torch/serialization.py
@@ -1119,7 +1119,7 @@ def load(
     map_location: MAP_LOCATION = None,
     pickle_module: Any = None,
     *,
-    weights_only: Optional[bool] = None,
+    weights_only: Optional[bool] = True,
     mmap: Optional[bool] = None,
     **pickle_load_args: Any,
 ) -> Any:
