diff --git a/torch/_dynamo/utils.py b/torch/_dynamo/utils.py
index 67ca455..930da80 100644
--- a/torch/_dynamo/utils.py
+++ b/torch/_dynamo/utils.py
@@ -2055,6 +2055,9 @@ def preserve_rng_state():
         skip_frame_if_in_functorch_mode(rng_state)
         if torch.cuda.is_available():
             cuda_rng_state = torch.clone(torch.cuda.get_rng_state())
+        elif custom_backend_mod := getattr(torch, torch._C._get_privateuse1_backend_name(), None):
+            if custom_backend_mod.is_available():
+                gpu_rng_state = torch.clone(custom_backend_mod.get_rng_state())
     try:
         yield
     finally:
@@ -2062,6 +2065,9 @@ def preserve_rng_state():
             torch.random.set_rng_state(rng_state)
             if torch.cuda.is_available():
                 torch.cuda.set_rng_state(cuda_rng_state)  # type: ignore[possibly-undefined]
+            elif custom_backend_mod := getattr(torch, torch._C._get_privateuse1_backend_name(), None):
+                if custom_backend_mod.is_available():
+                    custom_backend_mod.set_rng_state(gpu_rng_state)
 
 
 def is_jit_model(model0):
