diff --git a/torch/_dynamo/trace_rules.py b/torch/_dynamo/trace_rules.py
index e52c985d409..0ccd5af8b81 100644
--- a/torch/_dynamo/trace_rules.py
+++ b/torch/_dynamo/trace_rules.py
@@ -1868,6 +1868,7 @@ torch_c_binding_in_graph_functions = {
         "torch.resolve_conj",
         "torch.resolve_neg",
         "torch.result_type",
+        "torch.rms_norm",
         "torch.rnn_relu_cell",
         "torch.rnn_relu",
         "torch.rnn_tanh_cell",
