diff --git a/torch/distributed/tensor/experimental/_attention.py b/torch/distributed/tensor/experimental/_attention.py
index 6cd0672..c66dba8 100644
--- a/torch/distributed/tensor/experimental/_attention.py
+++ b/torch/distributed/tensor/experimental/_attention.py
@@ -565,7 +565,7 @@ def _templated_ring_attention_backward(
                 query=q,
                 key=k,
                 value=v,
-                out=out_,
+                output=out_,
                 logsumexp=lse,
                 is_causal=is_causal_behavior.value,
                 **kwargs,
@@ -746,6 +746,39 @@ def _scaled_dot_product_ring_cudnn_attention(
         return_debug_mask=return_debug_mask,
         scale=scale,
     )
+
+
+def _scaled_dot_product_ring_attention_flash_musa(
+    mesh: DeviceMesh,
+    query: torch.Tensor,
+    key: torch.Tensor,
+    value: torch.Tensor,
+    attn_mask: Optional[torch.Tensor] = None,
+    dropout_p: float = 0.0,
+    is_causal: bool = False,
+    return_debug_mask: bool = False,
+    *,
+    scale: Optional[float] = None,
+) -> tuple[torch.Tensor, ...]:
+
+    if query.dim() == 3:
+        seq_dim = 1
+    else:
+        seq_dim = 2
+    group = mesh.get_group()
+    return _templated_ring_attention(
+        group,
+        seq_dim,
+        aten._scaled_dot_product_attention_flash_musa,
+        query=query,
+        key=key,
+        value=value,
+        attn_mask=attn_mask,
+        is_causal=is_causal,
+        dropout_p=dropout_p,
+        return_debug_mask=return_debug_mask,
+        scale=scale,
+    )
 
 
 def _scaled_dot_product_ring_flash_attention_backward(
@@ -881,6 +914,48 @@ def _scaled_dot_product_ring_cudnn_attention_backward(
     )
 
 
+def _scaled_dot_product_ring_attention_flash_musa_backward(
+    mesh: DeviceMesh,
+    grad_out: torch.Tensor,
+    query: torch.Tensor,
+    key: torch.Tensor,
+    value: torch.Tensor,
+    output: torch.Tensor,
+    logsumexp: torch.Tensor,
+    dropout_mask: torch.Tensor,
+    rng_state: torch.Tensor,
+    dropout_p: float,
+    is_causal: bool = False,
+    attn_mask: Optional[torch.Tensor] = None,
+    *,
+    scale: Optional[float] = None,
+) -> tuple[torch.Tensor, ...]:
+
+    if query.dim() == 3:
+        seq_dim = 1
+    else:
+        seq_dim = 2
+    group = mesh.get_group()
+    return _templated_ring_attention_backward(
+        group,
+        seq_dim,
+        aten._scaled_dot_product_attention_flash_musa_backward.default,
+        grad_out=grad_out,
+        grad_out_name="grad_out",
+        query=query,
+        key=key,
+        value=value,
+        out=output,
+        logsumexp=logsumexp,
+        dropout_mask=dropout_mask,
+        rng_state=rng_state,
+        dropout_p=dropout_p,
+        is_causal=is_causal,
+        attn_mask=attn_mask,
+        scale=scale,
+    )
+
+
 def _sdpa_handler(
     op_call: torch._ops.OpOverload,
     args: tuple[object, ...],
@@ -903,9 +978,11 @@ def _sdpa_handler(
         aten._scaled_dot_product_flash_attention.default: _scaled_dot_product_ring_flash_attention,
         aten._scaled_dot_product_efficient_attention.default: _scaled_dot_product_ring_efficient_attention,
         aten._scaled_dot_product_cudnn_attention.default: _scaled_dot_product_ring_cudnn_attention,
+        aten._scaled_dot_product_attention_flash_musa.default: _scaled_dot_product_ring_attention_flash_musa,
         aten._scaled_dot_product_flash_attention_backward.default: _scaled_dot_product_ring_flash_attention_backward,
         aten._scaled_dot_product_efficient_attention_backward.default: _scaled_dot_product_ring_efficient_attention_backward,
         aten._scaled_dot_product_cudnn_attention_backward.default: _scaled_dot_product_ring_cudnn_attention_backward,
+        aten._scaled_dot_product_attention_flash_musa_backward.default: _scaled_dot_product_ring_attention_flash_musa_backward,
     }
     if op_call in call_maps:
         local_results = call_maps[op_call](
@@ -928,6 +1005,8 @@ customized_ops = {
     aten._scaled_dot_product_efficient_attention_backward.default: _sdpa_handler,
     aten._scaled_dot_product_cudnn_attention.default: _sdpa_handler,
     aten._scaled_dot_product_cudnn_attention_backward.default: _sdpa_handler,
+    aten._scaled_dot_product_attention_flash_musa.default: _sdpa_handler,
+    aten._scaled_dot_product_attention_flash_musa_backward.default: _sdpa_handler,
 }
 
 
