diff --git a/torch/_inductor/codegen/cpp_wrapper_cpu.py b/torch/_inductor/codegen/cpp_wrapper_cpu.py
index b04baf0..a6913bd 100644
--- a/torch/_inductor/codegen/cpp_wrapper_cpu.py
+++ b/torch/_inductor/codegen/cpp_wrapper_cpu.py
@@ -817,8 +817,8 @@ class CppWrapperCpu(PythonWrapperCodegen):
                 )
                 self.write_input_output_info("inputs_info_", idx, name)
 
-            all_cuda = all(
-                V.graph.get_original_value_of_constant(name).is_cuda
+            all_on_device = all(
+                V.graph.get_original_value_of_constant(name).device.type == self.device
                 for name in V.graph.constants
                 if name not in V.graph.folded_constants
             )
@@ -854,7 +854,7 @@ class CppWrapperCpu(PythonWrapperCodegen):
                     else tensor.untyped_storage().nbytes()
                 )
                 self.prefix.writeline(
-                    f"constants_info_[{idx}].data_size = {data_size if all_cuda else _align(data_size)};"
+                    f"constants_info_[{idx}].data_size = {data_size if all_on_device else _align(data_size)};"
                 )
 
                 from_folded = "true" if name in V.graph.folded_constants else "false"
