diff --git a/torch/storage.py b/torch/storage.py
index f65c080..97059d6 100644
--- a/torch/storage.py
+++ b/torch/storage.py
@@ -642,6 +642,8 @@ def _reset_warn_typed_storage_removal():
 
 
 def _get_device_from_module(module: str):
+    if torch._C._get_privateuse1_backend_name() in module:
+        return torch._C._get_privateuse1_backend_name()
     last_part = module.rsplit(".", 1)[-1]
     if last_part in ["cuda", torch._C._get_privateuse1_backend_name(), "hpu"]:
         return last_part
