diff --git a/torch/_utils.py b/torch/_utils.py
index 449075a..d05ec7e 100644
--- a/torch/_utils.py
+++ b/torch/_utils.py
@@ -98,6 +98,11 @@ def _to(self, device, non_blocking=False):
             assert not self.is_sparse, (
                 f"sparse storage is not supported for {device.type.upper()} tensors"
             )
+            import os
+            cpu_unified_flag = os.getenv("CPU_UNIFIED_FLAG", "False") == "True"
+            if(cpu_unified_flag):
+                self.edit_device(device)
+                return self
             untyped_storage = torch.UntypedStorage(self.size(), device=device)
             untyped_storage.copy_(self, non_blocking)
             return untyped_storage
