diff --git a/torch/autograd/profiler.py b/torch/autograd/profiler.py
index 9c5f8bd..a3faef3 100644
--- a/torch/autograd/profiler.py
+++ b/torch/autograd/profiler.py
@@ -10,6 +10,8 @@ from torch._C._profiler import _ExperimentalConfig
 
 from torch.autograd import (
     _disable_profiler,
+    _is_prepare_profiler_succeed,
+    _finish_profiler,
     _enable_profiler,
     _kineto_step,
     _prepare_profiler,
@@ -285,6 +287,8 @@ class profile:
 
     def _start_trace(self):
         self.entered = True
+        if not _is_prepare_profiler_succeed():
+            return
         _run_on_profiler_start()
         _enable_profiler(self.config(), self.kineto_activities)
 
@@ -293,6 +297,8 @@ class profile:
             return
         if self.use_cuda:
             torch.cuda.synchronize()
+        if not _is_prepare_profiler_succeed():
+            return
         self.kineto_results = _disable_profiler()
         _run_on_profiler_stop()
         parsed_results = self._parse_kineto_results(self.kineto_results)
@@ -345,9 +351,12 @@ class profile:
     table.__doc__ = EventList.table.__doc__
 
     def export_chrome_trace(self, path):
-        self._check_finish()
+        if _is_prepare_profiler_succeed():
+            self._check_finish()
         if kineto_available():
-            self.kineto_results.save(path)  # type: ignore[union-attr]
+            if _is_prepare_profiler_succeed():
+                self.kineto_results.save(path)  # type: ignore[union-attr]
+                _finish_profiler()
         else:
             return self.function_events.export_chrome_trace(path)  # type: ignore[union-attr]
 

