diff --git a/torch/csrc/inductor/aoti_torch/c/shim.h b/torch/csrc/inductor/aoti_torch/c/shim.h
index f56f6ec..ea7b2ba 100644
--- a/torch/csrc/inductor/aoti_torch/c/shim.h
+++ b/torch/csrc/inductor/aoti_torch/c/shim.h
@@ -686,36 +686,36 @@ AOTI_TORCH_EXPORT AOTITorchError aoti_torch_call_dispatcher(
     const char* overloadName,
     StableIValue* stack);
 
-#ifdef USE_CUDA
+#if 1
 
-struct CUDAGuardOpaque;
-using CUDAGuardHandle = CUDAGuardOpaque*;
+struct MUSAGuardOpaque;
+using MUSAGuardHandle = MUSAGuardOpaque*;
 
-AOTI_TORCH_EXPORT AOTITorchError aoti_torch_create_cuda_guard(
+AOTI_TORCH_EXPORT AOTITorchError aoti_torch_create_musa_guard(
     int32_t device_index,
-    CUDAGuardHandle* ret_guard // returns new reference
+    MUSAGuardHandle* ret_guard // returns new reference
 );
 
 AOTI_TORCH_EXPORT AOTITorchError
-aoti_torch_delete_cuda_guard(CUDAGuardHandle guard);
+aoti_torch_delete_musa_guard(MUSAGuardHandle guard);
 
 AOTI_TORCH_EXPORT AOTITorchError
-aoti_torch_cuda_guard_set_index(CUDAGuardHandle guard, int32_t device_index);
+aoti_torch_musa_guard_set_index(MUSAGuardHandle guard, int32_t device_index);
 
-struct CUDAStreamGuardOpaque;
-using CUDAStreamGuardHandle = CUDAStreamGuardOpaque*;
+struct MUSAStreamGuardOpaque;
+using MUSAStreamGuardHandle = MUSAStreamGuardOpaque*;
 
-AOTI_TORCH_EXPORT AOTITorchError aoti_torch_create_cuda_stream_guard(
+AOTI_TORCH_EXPORT AOTITorchError aoti_torch_create_musa_stream_guard(
     void* stream,
     int32_t device_index,
-    CUDAStreamGuardHandle* ret_guard // returns new reference
+    MUSAStreamGuardHandle* ret_guard // returns new reference
 );
 
 AOTI_TORCH_EXPORT AOTITorchError
-aoti_torch_delete_cuda_stream_guard(CUDAStreamGuardHandle guard);
+aoti_torch_delete_musa_stream_guard(MUSAStreamGuardHandle guard);
 
 AOTI_TORCH_EXPORT AOTITorchError
-aoti_torch_get_current_cuda_stream(int32_t device_index, void** ret_stream);
+aoti_torch_get_current_musa_stream(int32_t device_index, void** ret_stream);
 
 #endif // USE_CUDA
 
