diff --git a/torch/csrc/utils.h b/torch/csrc/utils.h
index 0fa7584..dc464b8 100644
--- a/torch/csrc/utils.h
+++ b/torch/csrc/utils.h
@@ -122,4 +122,4 @@ void storage_fill(const at::Storage& self, uint8_t value);
 void storage_set(const at::Storage& self, ptrdiff_t idx, uint8_t value);
 uint8_t storage_get(const at::Storage& self, ptrdiff_t idx);
 
-std::string uuid_to_string(const char* uuid_bytes);
+TORCH_PYTHON_API std::string uuid_to_string(const char* uuid_bytes);
