diff --git a/torch/csrc/profiler/python/combined_traceback.h b/torch/csrc/profiler/python/combined_traceback.h
index 0dcb9a1..f71033f 100644
--- a/torch/csrc/profiler/python/combined_traceback.h
+++ b/torch/csrc/profiler/python/combined_traceback.h
@@ -11,7 +11,7 @@ namespace torch {
 // returns std::vector because one use is to call it with a batch of
 // tracebacks that come from a larger datastructure (e.g. a memory snapshot)
 // and then have more c++ code to put those objects in the right place.
-std::vector<pybind11::object> py_symbolize(
+TORCH_API std::vector<pybind11::object> py_symbolize(
     std::vector<CapturedTraceback*>& to_symbolize);
 
 // requires GIL to be held, frees any pending free frames
