diff --git a/torch/nn/modules/__init__.py b/torch/nn/modules/__init__.py
index 99260ad43fc..b4d5814620d 100644
--- a/torch/nn/modules/__init__.py
+++ b/torch/nn/modules/__init__.py
@@ -13,6 +13,8 @@ from .activation import (
     LogSigmoid,
     LogSoftmax,
     Mish,
+    RoPE,
+    SwishGLU,
     MultiheadAttention,
     PReLU,
     ReLU,
@@ -297,6 +299,7 @@ __all__ = [
     "ReplicationPad1d",
     "ReplicationPad2d",
     "ReplicationPad3d",
+    "RoPE",
     "SELU",
     "Sequential",
     "SiLU",
@@ -309,6 +312,7 @@ __all__ = [
     "Softplus",
     "Softshrink",
     "Softsign",
+    "SwishGLU",
     "SyncBatchNorm",
     "Tanh",
     "Tanhshrink",
