diff --git a/torch/csrc/distributed/c10d/TCPStoreLibUvBackend.cpp b/torch/csrc/distributed/c10d/TCPStoreLibUvBackend.cpp
index 0cac94b..6a1c419 100644
--- a/torch/csrc/distributed/c10d/TCPStoreLibUvBackend.cpp
+++ b/torch/csrc/distributed/c10d/TCPStoreLibUvBackend.cpp
@@ -7,6 +7,7 @@
 #include <unordered_set>
 #include <utility>
 #include <vector>
+#include <c10/util/Logging.h>
 
 #include <c10/util/thread_name.h>
 #include <fmt/format.h>
@@ -1060,6 +1061,7 @@ void LibUVStoreDaemon::init(const TCPStoreOptions& opts) {
       port_,
       ", expected to be bound to port ",
       opts.port);
+  LOG(INFO) << "libuv backend is init success!!!!!";
 }
 
 LibUVStoreDaemon::LibUVStoreDaemon(int port) : port_(port) {
