diff --git a/torch/_inductor/scheduler.py b/torch/_inductor/scheduler.py
index 349e400..eed040e 100644
--- a/torch/_inductor/scheduler.py
+++ b/torch/_inductor/scheduler.py
@@ -3953,7 +3953,7 @@ class Scheduler:
                 and (device_props := torch.cuda.get_device_properties(device)).major < 7
             ):
                 raise GPUTooOldForTriton(device_props, inspect.currentframe())
-            elif is_gpu(device.type) and not device.type == "mps":
+            elif is_gpu(device.type) and not device.type == "musa":
                 raise TritonMissing(inspect.currentframe())
 
         return device_scheduling(self)
