diff --git a/torch/_inductor/codecache.py b/torch/_inductor/codecache.py
index 772536ffb34..56ea9af22f3 100644
--- a/torch/_inductor/codecache.py
+++ b/torch/_inductor/codecache.py
@@ -155,6 +155,8 @@ def get_kernel_bin_format(device: str) -> str:
         return "cubin" if torch.version.hip is None else "hsaco"
     elif device == "xpu":
         return XPU_KERNEL_FORMAT
+    elif device == "musa":
+        return "mubin"
     else:
         return ""
 
@@ -203,7 +205,7 @@ class CacheBase:
             else:
                 system["device"]["name"] = device_properties.gcnArchName
                 system["version"]["hip"] = torch.version.hip
-        except (AssertionError, RuntimeError):
+        except (AssertionError, RuntimeError, AttributeError):
             # If cuda is not installed, none of the above config is relevant.
             system = {}
 
@@ -1749,7 +1751,7 @@ class CudaKernelParamCache:
             specified_dir=split_aot_inductor_output_path(
                 config.aot_inductor.output_path
             )[0],
-            key=basename,
+            # key=basename,
         )
         # Retrieve the basename again in case it is a generated hashcode
         basename, _ = get_name_and_dir_from_output_file_path(bin_path)
@@ -1831,7 +1833,12 @@ class AotCodeCompiler:
         """
         generated_files: list[str | Weights] = additional_files  # type: ignore[assignment]
 
-        _set_gpu_runtime_env()  # cpp_extension consults the env
+        if device_type == "musa":
+            from torch_musa.utils.musa_extension import _find_musa_home
+
+            _find_musa_home()  # cpp_extension consults the env
+        else:
+            _set_gpu_runtime_env()  # cpp_extension consults the env
 
         picked_vec_isa = pick_vec_isa()
         vec_isa_cmd_gen = CppBuilder(
@@ -2012,7 +2019,7 @@ class AotCodeCompiler:
 
             # Intel compiler failed to compile this manually constructed assembly file.
             # Switch XPU to use consts cpp build.
-            if device_type == "xpu":
+            if device_type in ("xpu", "musa"):
                 use_asm_build = False
 
             is_large_consts = len(consts) > 1024
@@ -2228,13 +2235,13 @@ end
                 if specified_artifact_name
                 else str(wrapper_path_operator.with_suffix(".so"))
             )
-            all_cuda = all(
-                graph.get_original_value_of_constant(name).is_cuda
+            all_on_device = all(
+                graph.get_original_value_of_constant(name).device.type == device_type
                 for name in graph.constants
                 if name not in graph.folded_constants
             )
 
-            def _to_bytes(t: torch.Tensor, all_cuda: bool) -> bytes:
+            def _to_bytes(t: torch.Tensor, all_on_device: bool) -> bytes:
                 def _pad_to_alignment(raw_bytes: bytes) -> bytes:
                     padded_bytes = raw_bytes.ljust(
                         (len(raw_bytes) + ALIGN_BYTES - 1) // ALIGN_BYTES * ALIGN_BYTES,
@@ -2263,14 +2270,14 @@ end
                 )
                 # pyrefly: ignore [missing-attribute]
                 raw_bytes = bytes(raw_array.contents)
-                return raw_bytes if all_cuda else _pad_to_alignment(raw_bytes)
+                return raw_bytes if all_on_device else _pad_to_alignment(raw_bytes)
 
             if (
                 config.aot_inductor.package_constants_in_so
                 or config.aot_inductor.package_constants_on_disk_format == "binary_blob"
             ):
                 serialized_weights = b"".join(
-                    _to_bytes(graph.get_original_value_of_constant(name), all_cuda)
+                    _to_bytes(graph.get_original_value_of_constant(name), all_on_device)
                     for name in graph.constants
                     if name not in graph.folded_constants
                 )
@@ -2591,10 +2598,10 @@ end
                         f_weights.write(struct.pack("q", magic_number))
 
                     generated_files.append(weight_file)
-                else:
-                    # TODO: unify to always use mmap_weights
-                    generated_files.append(consts_o)
-                    so_builder.save_src_to_cmake(cmake_path, consts_o)
+
+                # TODO: unify to always use mmap_weights
+                generated_files.append(consts_o)
+                so_builder.save_src_to_cmake(cmake_path, consts_o)
 
                 # Different CMake strategies for CUDA vs ROCm:
                 # - CUDA: Save asm for CMake to recompile (user has nvcc)
