diff --git a/torch/nn/functional.pyi.in b/torch/nn/functional.pyi.in
index 1a72b0f..24a3eee 100644
--- a/torch/nn/functional.pyi.in
+++ b/torch/nn/functional.pyi.in
@@ -599,6 +599,14 @@ def multi_head_attention_forward(
     average_attn_weights: bool = True,
     is_causal: bool = False,
 ) -> tuple[Tensor, Optional[Tensor]]: ...
+def rope(
+    input: Tensor,
+    freq_cis: Tensor,
+    rotary_interleaved: bool,
+    batch_first: bool,
+    multi_latent_attention: bool,
+) -> Tensor: ...
+def swish_glu(input: Tensor) -> Tensor: ...
 
 ${imported_hints}
 
