diff --git a/torch/nn/modules/__init__.py b/torch/nn/modules/__init__.py
index 99260ad..506c8e0 100644
--- a/torch/nn/modules/__init__.py
+++ b/torch/nn/modules/__init__.py
@@ -30,6 +30,8 @@ from .activation import (
     Tanh,
     Tanhshrink,
     Threshold,
+    RoPE,
+    SwishGLU,
 )
 from .adaptive import AdaptiveLogSoftmaxWithLoss
 from .batchnorm import (
@@ -297,6 +299,7 @@ __all__ = [
     "ReplicationPad1d",
     "ReplicationPad2d",
     "ReplicationPad3d",
+    "RoPE",
     "SELU",
     "Sequential",
     "SiLU",
@@ -309,6 +312,7 @@ __all__ = [
     "Softplus",
     "Softshrink",
     "Softsign",
+    "SwishGLU",
     "SyncBatchNorm",
     "Tanh",
     "Tanhshrink",
