diff --git a/torch/_inductor/runtime/triton_heuristics.py b/torch/_inductor/runtime/triton_heuristics.py
index 2a1447fbf0b..9a9f422f67e 100644
--- a/torch/_inductor/runtime/triton_heuristics.py
+++ b/torch/_inductor/runtime/triton_heuristics.py
@@ -1240,7 +1240,7 @@ class CachingAutotuner(KernelInterface):
         from torch._inductor import config
         from torch._inductor.codecache import CudaKernelParamCache
 
-        bin_type = {"hip": "hsaco", "xpu": XPU_KERNEL_FORMAT}.get(
-            self.device_props.type, "cubin"
+        bin_type = {"hip": "hsaco", "xpu": XPU_KERNEL_FORMAT, "musa": "mubin"}.get(
+            self.device_props.type, "mubin"
         )
         binary = launcher.bin.asm[bin_type]
@@ -1265,7 +1265,7 @@ class CachingAutotuner(KernelInterface):
 
         # Everything else: capture architecture-specific assembly
         else:
-            asm_type = {"hip": "amdgcn", "cuda": "ptx", "xpu": "spv"}.get(
+            asm_type = {"hip": "amdgcn", "cuda": "ptx", "xpu": "spv", "musa": "llir"}.get(
                 self.device_props.type
             )
             asm = launcher.bin.asm.get(asm_type, None)
@@ -2464,14 +2464,14 @@ def triton_config(
         num_warps = _num_warps(
             conditional_product(x, y, z) // num_elements_per_warp, min_num_warps=1
         )
-    # we are going to arrive at 2 warps only if bs was too small due to
-    # numel being too small. However to workaround some ptx bugs we still
-    # want at least 4 warps if there's enough elements per thread
-    # given that this is a rare situation, don't expect this to affect perf
-    # in general
-    # see https://github.com/pytorch/pytorch/pull/97950
-    if conditional_product(x, y, z) >= 128 and not torch.version.hip:
-        num_warps = max(num_warps, 4)
+        # we are going to arrive at 2 warps only if bs was too small due to
+        # numel being too small. However to workaround some ptx bugs we still
+        # want at least 4 warps if there's enough elements per thread
+        # given that this is a rare situation, don't expect this to affect perf
+        # in general
+        # see https://github.com/pytorch/pytorch/pull/97950
+        if conditional_product(x, y, z) >= 128 and not torch.version.hip:
+            num_warps = max(num_warps, 4)
     xnumel = size_hints["x"]
     ynumel = size_hints.get("y")
     znumel = size_hints.get("z")
@@ -2878,6 +2878,13 @@ def pointwise(
                 ),
                 *hinted_configs,
             ]
+            if inductor_meta.get("max_autotune") or inductor_meta.get("max_autotune_pointwise"):
+                for num_warps in [1, 2, 4, 8]:
+                    for _bs in [bs, 64, 128, 256]:
+                        for nepw in [128, 256, 512]:
+                            configs.append(triton_config_with_settings(
+                                size_hints, _bs,  num_elements_per_warp=nepw, num_warps=num_warps))
+
             # Additional configs appended for ROCm builds
             if torch.version.hip:
                 if inductor_meta.get("max_autotune_pointwise"):
@@ -3244,6 +3251,10 @@ def _reduction_configs(
         contiguous_config,
         outer_config,
         tiny_config,
+        make_config(16, 16),
+        make_config(32, 32),
+        make_config(32, 128),
+        make_config(128, 128),
         make_config(64, 64),
         make_config(8, 512),
         # halve the XBLOCK/Rn_BLOCK compared to outer_config
@@ -3260,7 +3271,23 @@ def _reduction_configs(
             ]
         )
 
-    return result_configs
+    if not inductor_meta.get("max_autotune"):
+        return result_configs
+
+    ret = []
+    for cfg in result_configs:
+        for num_warps in [1, 2, 4, 8]:
+            ret.append(
+                Config(
+                    cfg.kwargs,
+                    num_warps=num_warps,
+                    num_stages=cfg.num_stages,
+                    num_ctas=cfg.num_ctas,
+                    maxnreg=cfg.maxnreg,
+                    pre_hook=cfg.pre_hook,
+                )
+            )
+    return ret
 
 
 def match_target_block_product(
