diff --git a/torch/nn/functional.pyi.in b/torch/nn/functional.pyi.in
index ed68b8e9b7a..47ce3b55f44 100644
--- a/torch/nn/functional.pyi.in
+++ b/torch/nn/functional.pyi.in
@@ -602,6 +602,13 @@ def multi_head_attention_forward(
     average_attn_weights: bool = True,
     is_causal: bool = False,
 ) -> Tuple[Tensor, Optional[Tensor]]: ...
+def rope(
+    input: Tensor,
+    freq_cis: Tensor,
+    rotary_interleaved: bool,
+    batch_first: bool,
+) -> Tensor: ...
+def swish_glu(input: Tensor) -> Tensor: ...
 
 ${imported_hints}
 
