diff --git a/torch/_dynamo/utils.py b/torch/_dynamo/utils.py
index c30f5ce..96bba9e 100644
--- a/torch/_dynamo/utils.py
+++ b/torch/_dynamo/utils.py
@@ -698,14 +698,18 @@ def preserve_rng_state():
     with torch.utils._python_dispatch._disable_current_modes():
         rng_state = torch.clone(torch.random.get_rng_state())
         if torch.cuda.is_available():
-            cuda_rng_state = torch.clone(torch.cuda.get_rng_state())
+            gpu_rng_state = torch.clone(torch.cuda.get_rng_state())
+        elif hasattr(torch, "musa") and torch.musa.is_available():
+            gpu_rng_state = torch.clone(torch.musa.get_rng_state())
     try:
         yield
     finally:
         with torch.utils._python_dispatch._disable_current_modes():
             torch.random.set_rng_state(rng_state)
             if torch.cuda.is_available():
-                torch.cuda.set_rng_state(cuda_rng_state)
+                torch.cuda.set_rng_state(gpu_rng_state)
+            elif hasattr(torch, "musa") and torch.musa.is_available():
+                torch.musa.set_rng_state(gpu_rng_state)
 
 
 def is_jit_model(model0):
