diff --git a/torch/ao/nn/intrinsic/qat/modules/__init__.py b/torch/ao/nn/intrinsic/qat/modules/__init__.py
index f44820c637e..89aeb6e2130 100644
--- a/torch/ao/nn/intrinsic/qat/modules/__init__.py
+++ b/torch/ao/nn/intrinsic/qat/modules/__init__.py
@@ -7,9 +7,11 @@ from .conv_fused import (
     ConvBnReLU1d,
     ConvBnReLU2d,
     ConvBnReLU3d,
+    ConvBnSiLU2d,
     ConvReLU1d,
     ConvReLU2d,
     ConvReLU3d,
+    ConvSiLU2d,
     update_bn_stats,
     freeze_bn_stats,
 )
@@ -20,12 +22,14 @@ __all__ = [
     "ConvReLU1d",
     "ConvReLU2d",
     "ConvReLU3d",
+    "ConvSiLU2d",
     "ConvBn1d",
     "ConvBn2d",
     "ConvBn3d",
     "ConvBnReLU1d",
     "ConvBnReLU2d",
     "ConvBnReLU3d",
+    "ConvBnSiLU2d",
     "update_bn_stats",
     "freeze_bn_stats",
 ]
