diff --git a/torch/cuda/_utils.py b/torch/cuda/_utils.py
index 1794ca9..b14d39e 100644
--- a/torch/cuda/_utils.py
+++ b/torch/cuda/_utils.py
@@ -28,12 +28,12 @@ def _get_device_index(
         device = torch.device(device)
     if isinstance(device, torch.device):
         if allow_cpu:
-            if device.type not in ["cuda", "cpu"]:
-                raise ValueError(f"Expected a cuda or cpu device, but got: {device}")
-        elif device.type != "cuda":
-            raise ValueError(f"Expected a cuda device, but got: {device}")
+            if device.type not in ["musa", "cpu"]:
+                raise ValueError(f"Expected a musa or cpu device, but got: {device}")
+        elif device.type != "musa":
+            raise ValueError(f"Expected a musa device, but got: {device}")
     if not torch.jit.is_scripting():
-        if isinstance(device, torch.cuda.device):
+        if isinstance(device, torch.musa.device):
             return device.idx
     return _torch_get_device_index(device, optional, allow_cpu)
 
