diff --git a/torch/_functorch/partitioners.py b/torch/_functorch/partitioners.py
index b0f609d..4d6b805 100644
--- a/torch/_functorch/partitioners.py
+++ b/torch/_functorch/partitioners.py
@@ -639,6 +639,9 @@ def functionalize_rng_ops(
     # Unique id to generate name
     uid = itertools.count()
 
+    # for external backend
+    custom_backend_name = torch._C._get_privateuse1_backend_name()
+
     def get_rng_ops(gmod):
         random_nodes = {}
         for node in gmod.graph.nodes:
@@ -665,12 +668,16 @@ def functionalize_rng_ops(
             if isinstance(candidate, torch.Tensor):
                 if candidate.device.type == "cuda":
                     return "cuda"
+                elif candidate.device.type == custom_backend_name:
+                    return custom_backend_name
 
         return "cpu"
 
     def get_sample_rng_state(device):
         if device == "cuda":
             return torch.cuda.get_rng_state()
+        elif device == custom_backend_name:
+            return getattr(torch, custom_backend_name).get_rng_state()
         return torch.get_rng_state()
 
     # Step 1 - Construct a mapping of rng node between the fwd and its counterpart in bwd.
