diff --git a/torch/ao/nn/intrinsic/__init__.py b/torch/ao/nn/intrinsic/__init__.py
index a18bae3..9a7da9a 100644
--- a/torch/ao/nn/intrinsic/__init__.py
+++ b/torch/ao/nn/intrinsic/__init__.py
@@ -12,9 +12,11 @@ __all__ = [
     'ConvBnReLU1d',
     'ConvBnReLU2d',
     'ConvBnReLU3d',
+    'ConvBnSiLU2d',
     'ConvReLU1d',
     'ConvReLU2d',
     'ConvReLU3d',
+    'ConvSiLU2d',
     'LinearReLU',
     'BNReLU2d',
     'BNReLU3d',
@@ -23,6 +25,7 @@ __all__ = [
     'LinearTanh',
     'ConvAdd2d',
     'ConvAddReLU2d',
+    'ConvAddSiLU2d',
 ]
 
 # We are exposing all subpackages to the end-user.
