diff --git a/torch/_inductor/codecache.py b/torch/_inductor/codecache.py
index dc9f5c2..43a86b7 100644
--- a/torch/_inductor/codecache.py
+++ b/torch/_inductor/codecache.py
@@ -187,7 +187,7 @@ class CacheBase:
             else:
                 system["device"]["name"] = device_properties.gcnArchName
                 system["version"]["hip"] = torch.version.hip
-        except (AssertionError, RuntimeError):
+        except (AssertionError, RuntimeError, AttributeError):
             # If cuda is not installed, none of the above config is relevant.
             system = {}
 
