diff --git a/torch/nn/functional.pyi.in b/torch/nn/functional.pyi.in
index 4955aef0524..75d2a6109c8 100644
--- a/torch/nn/functional.pyi.in
+++ b/torch/nn/functional.pyi.in
@@ -292,6 +292,12 @@ def layer_norm(
     bias: Optional[Tensor] = ...,
     eps: float = ...,
 ) -> Tensor: ...
+def rms_norm(
+    input: Tensor,
+    normalized_shape: Sequence[int],
+    weight: Optional[Tensor] = ...,
+    eps: Optional[float] = ...,
+) -> Tensor: ...
 def group_norm(
     input: Tensor,
     num_groups: int,
