diff --git a/torch/_dynamo/variables/torch.py b/torch/_dynamo/variables/torch.py
index 694edb4c691..2d59a3be144 100644
--- a/torch/_dynamo/variables/torch.py
+++ b/torch/_dynamo/variables/torch.py
@@ -526,6 +526,14 @@ class TorchCtxManagerClassVariable(BaseTorchVariable):
             torch.amp.autocast_mode.autocast,
             torch.cuda.amp.autocast,
             torch.cpu.amp.autocast,
+        ) or self.value is getattr(
+            getattr(
+                getattr(torch, torch._C._get_privateuse1_backend_name(), None),
+                "amp",
+                None,
+            ),
+            "autocast",
+            None,
         ):
             # pyrefly: ignore [bad-argument-type]
             return AutocastModeVariable.create(self.value, args, kwargs)
