diff --git a/torch/csrc/profiler/kineto_shim.h b/torch/csrc/profiler/kineto_shim.h
index e92cbf0..6bc19a6 100644
--- a/torch/csrc/profiler/kineto_shim.h
+++ b/torch/csrc/profiler/kineto_shim.h
@@ -140,6 +140,12 @@ namespace autograd {
 namespace profiler {
 c10::DeviceType deviceTypeFromActivity(libkineto::ActivityType activity_type);
 
+TORCH_API bool isSyncProfilingRunning();
+// Sync api profiling is finished, tell kineto this good news,
+// kineto may launch a pending on-demand profiling if exists.
+// Otherwise, on-demand profiling will be blocked forever.
+TORCH_API void setSyncProfilingRunningFalse();
+
 TORCH_API void addMetadataJson(
     const std::string& key,
     const std::string& value);

