diff --git a/torch/csrc/autograd/profiler_python.cpp b/torch/csrc/autograd/profiler_python.cpp
index da1cedf..66f5d21 100644
--- a/torch/csrc/autograd/profiler_python.cpp
+++ b/torch/csrc/autograd/profiler_python.cpp
@@ -966,7 +966,12 @@ class PostProcess {
     using stack_t = std::vector<std::shared_ptr<Result>>;
     const auto initial_size = out.size();
     auto pop = [](stack_t& stack, c10::time_t t) {
-      TORCH_INTERNAL_ASSERT(!stack.empty(), "Python replay stack is empty.");
+      // musaStub is not available hence there is no method to record replay
+      // TORCH_INTERNAL_ASSERT(stack.size(), "Python replay stack is empty.");
+      if (stack.empty()) {
+        LOG(INFO) << "Python replay stack is empty.";
+        return;
+      }
       std::get<ExtraFields<E>>(stack.back()->extra_fields_).end_time_ns_ = t;
       stack.pop_back();
     };
