diff --git a/torch/autograd/__init__.py b/torch/autograd/__init__.py
index 3993f78..4504110 100644
--- a/torch/autograd/__init__.py
+++ b/torch/autograd/__init__.py
@@ -470,6 +470,8 @@ if not torch._C._autograd_init():
 from torch._C._autograd import (
     _add_metadata_json,
     _disable_profiler,
+    _is_prepare_profiler_succeed,
+    _finish_profiler,
     _disable_profiler_legacy,
     _enable_profiler,
     _enable_profiler_legacy,
@@ -492,7 +494,7 @@ from torch._C._autograd import (
     SavedTensor,
 )
 
-from torch._C._profiler import ProfilerActivity, ProfilerConfig, ProfilerState
+from torch._C._profiler import ProfilerActivity, ProfilerConfig, ProfilerState, RecordScope
 
 from . import profiler
 

